### PR TITLE
Add `TcpStream::set_linger` and `TcpStream::linger`

### PR DESCRIPTION
**Pull request by ibraheemdev**
Adds methods for getting/setting the `SO_LINGER` option on TCP sockets. Behavior is consistent across Unix and Windows.

r? @joshtriplett (I noticed you've been reviewing net related PRs)

### CODE REVIEW

File: library/std/src/sys/unix/net.rs [hunk 1]
```diff
@@ -376,6 +384,21 @@ impl Socket {
         Ok(())
     }
 
+    pub fn set_linger(&self, linger: Option<Duration>) -> io::Result<()> {
+        let linger = libc::linger {
+            l_onoff: linger.is_some() as libc::c_int,
+            l_linger: linger.map(|dur| dur.as_secs() as libc::c_int).unwrap_or_default(),
```
Comment (joshtriplett) [inline library/std/src/sys/unix/net.rs:390]: `Duration` implements `Default`:

```suggestion
            l_linger: linger.unwrap_or_default().as_secs() as libc::c_int,
```

**Comment by joshtriplett** [comment]
LGTM, r=me once CI passes.

**Comment by ibraheemdev** [comment]
CI passed.

File: library/std/src/lib.rs [hunk 2]
```diff
@@ -321,6 +321,7 @@
 #![feature(stdsimd)]
 #![feature(stmt_expr_attributes)]
 #![feature(str_internals)]
+#![feature(tcp_linger)]
```
Comment (m-ou-se) [inline library/std/src/lib.rs:324]: `std` doesn't need to enable its own features.

```suggestion
```

**Pull request merged**